Rerun job fails if parents job with *_custom as target

Rerun job fails if parents job with *_custom as target. Job with [target as *_custom ( normally grinder job) can also trigger the rerun job](https://github.com/adoptium/aqa-tests/blob/master/buildenv/jenkins/JenkinsfileBase#L1329). But rerun will fail if custom_target is not the default.

 https://ci.adoptium.net/view/Test_grinder/job/Grinder/10380/
Target:hotspot_custom
CUSTOM_TARGET: compiler/arguments/CheckCICompilerCount.java compiler/c2/cr6340864/TestByteVect.java compiler/c2/cr6340864/TestDoubleVect.java compiler/c2/cr6340864/TestFloatVect.java compiler/c2/cr6340864/TestIntVect.java compiler/c2/cr6340864/TestIntVectRotate.java
rerun job https://ci.adoptium.net/job/Grinder_rerun/121/console
Failed 

```
10:17:34  make -f makeGen.mk AUTO_DETECT=true MODE=tests TESTTARGET=testList TESTLIST=HOTSPOT_CUSTOM_TARGET=compiler/arguments/CheckCICompilerCount.java compiler/c2/cr6340864/TestByteVect.java compiler/c2/cr6340864/TestDoubleVect.java compiler/c2/cr6340864/TestFloatVect.java compiler/c2/cr6340864/TestIntVect.java compiler/c2/cr6340864/TestIntVectRotate.java
10:17:34  make[1]: Entering directory '/cygdrive/f/workspace/workspace/Grinder_rerun/aqa-tests/TKG'
10:17:35  make[1]: *** No rule to make target 'compiler/c2/cr6340864/TestByteVect.java'.  Stop.
10:17:35  make[1]: Leaving directory '/cygdrive/f/workspace/workspace/Grinder_rerun/aqa-tests/TKG'
10:17:35  make: *** [makefile:69: _testList] Error 2
```

--- a/buildenv/jenkins/JenkinsfileBase
+++ b/buildenv/jenkins/JenkinsfileBase
@@ -1357,10 +1357,18 @@ def triggerRerunJob () {
 							if (env.RERUN_TESTCASES) {
 								childParams << string(name: param.key, value: env.RERUN_CUSTOMTARGET)
 							} else {
-								childParams << string(name: param.key, value: "testList TESTLIST=" + env.FAILED_TESTS)
+								if (env.FAILED_TESTS.contains("_custom")) {
+									childParams << string(name: param.key, value: env.FAILED_TESTS)
+								} else {
+									childParams << string(name: param.key, value: "testList TESTLIST=" + env.FAILED_TESTS)
+								}
 							}
 						} else if (param.key == "CUSTOM_TARGET") {
+							if (env.RERUN_TESTCASES) {
 								childParams << string(name: param.key, value: env.RERUN_TESTCASES)
+							} else {
+								childParams << string(name: param.key, value: param.value.toString())
+							}
 						} else if (param.key == "PARALLEL") {
 							childParams << string(name: param.key, value: "None")
 						} else if (param.key == "NUM_MACHINES") {

--- a/buildenv/jenkins/JenkinsfileBase
+++ b/buildenv/jenkins/JenkinsfileBase
@@ -1294,6 +1294,7 @@ def addFailedTestsGrinderLink(paths=""){
 				def testCases=customizedTestCases.values().first()
 				def customTarget=customizedTestCases.keySet().first()
 				def testCaseSize=testCases.count(' ')
+				// Limitation of Header Fields https://github.com/adoptium/aqa-tests/issues/5241#issuecomment-2161927014
 				if (testCaseSize < 80) {
 					env.RERUN_CUSTOMTARGET="${customTarget}_custom"
 					env.RERUN_TESTCASES="${testCases}"

--- a/buildenv/jenkins/JenkinsfileBase
+++ b/buildenv/jenkins/JenkinsfileBase
@@ -1267,6 +1267,7 @@ def addFailedTestsGrinderLink(paths=""){
 		if (failedTestList) {
 			String failedTests = failedTestList.substring(0, failedTestList.length() - 1)
 			env.FAILED_TESTS= failedTests ?: ""
+			env.RERUN_TESTCASES=""
 			failedTestList = "testList+TESTLIST=" + failedTests
 			String url = env.RERUN_LINK
 			def failedTestUrl = url.replace(env.FAILED_TEST_TARGET, "TARGET=$failedTestList")
@@ -1287,6 +1288,16 @@ def addFailedTestsGrinderLink(paths=""){
 			}
 			if (jckDevtoolsFailedTestCaseList) {
 				customizedTestCases['jckdevtools'] = "${jckDevtoolsFailedTestCaseList}"
+			}
+			
+			if (customizedTestCases.size() == 1) {
+				def testCases=customizedTestCases.values().first()
+				def customTarget=customizedTestCases.keySet().first()
+				def testCaseSize=testCases.count(' ')
+				if (testCaseSize < 80) {
+					env.RERUN_CUSTOMTARGET="${customTarget}_custom"
+					env.RERUN_TESTCASES="${testCases}"
+				}
 			}
 			customizedTestCases.each { target, testcases ->
 				def tempTestCases = testcases.substring(0, testcases.length() - 1)
@@ -1343,7 +1354,13 @@ def triggerRerunJob () {
 						// set PARALLEL, NUM_MACHINES and TEST_TIME to default values
 						// set TARGET to failed tests and set ITERATIONS to rerunIterations
 						if (param.key == "TARGET") {
-							childParams << string(name: param.key, value: "testList TESTLIST=" + env.FAILED_TESTS)
+							if (env.RERUN_TESTCASES) {
+								childParams << string(name: param.key, value: env.RERUN_CUSTOMTARGET)
+							} else {
+								childParams << string(name: param.key, value: "testList TESTLIST=" + env.FAILED_TESTS)
+							}
+						} else if (param.key == "CUSTOM_TARGET") {
+								childParams << string(name: param.key, value: env.RERUN_TESTCASES)
 						} else if (param.key == "PARALLEL") {
 							childParams << string(name: param.key, value: "None")
 						} else if (param.key == "NUM_MACHINES") {

